projects/igniteui-angular/src/lib/core/styles/components/tree/_tree-theme.scss

```diff
@@ -0,0 +1,356 @@
+
+    %node-wrapper,
+    %node-toggle-button,
+    %node-select {
+        align-items: center;
+    }
+
+    %node-toggle-button,
+    %node-select {
+        margin-#{$right}: $icon-space;
+    }
+
+    %node-content,
+    %node-toggle-button,
+    %node-select {
+        z-index: 1;
+    }
+
+    %node-toggle-button--hidden {
+        visibility: hidden;
+    }
+
+    %node-wrapper {
+        min-height: map-get($node-height, 'comfortable');
+        padding: 0 map-get($node-indent, 'comfortable');
+        position: relative;
+        background: --var($theme, 'background');
+        color: --var($theme, 'foreground');
+
+        // We need this here, since we count on it to calculate the width of the tree drop indicator
+        igx-icon {
+            width: $icon-size;
+            height: $icon-size;
+            font-size: $icon-size;
+        }
+
+        &::after {
+            content: '';
+            position: absolute;
+            top: 0;
+            left: 0;
+            width: 100%;
+            height: 100%;
+            background: transparent;
+            z-index: 0;
+        }
+
+        &:hover {
+            &::after {
+                background: --var($theme, '_hover-color');
+            }
+        }
+
+        &:focus {
+            outline-width: 0;
+        }
+
+        igx-circular-bar {
+            width: $icon-size;
+            height: $icon-size;
+        }
+
+        &--cosy {
+            min-height: map-get($node-height, 'cosy');
+            padding: 0 map-get($node-indent, 'cosy');
+
+            %node-spacer {
+                width: map-get($node-indent, 'cosy')
+            }
+
+            %node-drop-indicator {
+                #{$right}: map-get($node-indent, 'cosy');
+                width: map-get($drop-indicator-width, 'cosy');
+            }
+
+            igx-circular-bar {
+                width: calc(#{$icon-size} - 4px);
+                height: calc(#{$icon-size} - 4px);
+            }
+        }
+
+        &--compact {
+            min-height: map-get($node-height, 'compact');
+            padding: 0 map-get($node-indent, 'compact');
+
+            %node-spacer {
+                width: map-get($node-indent, 'compact')
+            }
+
+            %node-drop-indicator {
+                #{$right}: map-get($node-indent, 'compact');
+                width: map-get($drop-indicator-width, 'compact');
+            }
+
+            igx-circular-bar {
+                width: calc(#{$icon-size} - 6px);
+                height: calc(#{$icon-size} - 6px);
+            }
+        }
+    }
+
+    %node-wrapper--selected {
+        background: --var($theme, 'background-selected');
+        color: --var($theme, 'foreground-selected');
+    }
+
+    %node-wrapper--active {
+        background: --var($theme, 'background-active');
+        color: --var($theme, 'foreground-active');
+    }
+
+    %node-wrapper--selected,
+    %node-wrapper--active-selected {
+        &:hover {
+            &::after {
```

Comment:
Combine the hover and after selectors to reduce the nesting depth and remove the stylelint ignore rules.